framework/src/modules/auth/module.ts

```diff
@@ -182,20 +182,25 @@ export class AuthModule extends BaseModule {
 
 	public async beforeCommandExecute(context: TransactionExecuteContext): Promise<void> {
 		const { transaction } = context;
-		const store = this.stores.get(AuthAccountStore);
-		const senderExist = await store.has(context, transaction.senderAddress);
-		if (!senderExist) {
-			await store.set(context, context.transaction.senderAddress, {
+		const authAccountStore = this.stores.get(AuthAccountStore);
+		let senderAccount: AuthAccount;
+
+		try {
+			senderAccount = await authAccountStore.get(context, transaction.senderAddress);
```

Comment:
it would be even better if we encapsulate this logic into `getOrDefault` method in `AuthAccountStore`

Reply:
Good point 👍🏻 

Updated.